.github/workflows/coding-standards.yml

```
@@ -18,9 +18,10 @@ on:
       - '3.[89]'
       - '[4-9].[0-9]'
     paths:
-      # Any change to a PHP or JavaScript file should run checks.
-      - '**.js'
-      - '**.php'
+      # Any change to a relevant PHP or JavaScript file should run checks.
+      - 'src/**.js'
+      - 'src/**.php'
+      - 'tests/**.php'
```

Misses files in the root directory, currently the PHP sample configs, webpack.

Which files in particular are you trying to ignore here? 

:1234: This applies to other changes too, so I won't repeat myself.

---

For example, with this change the workflow wouldn't unnecessarily run when an e2e, performance, or QUnit test file is changed (which are all `.js` files).

---

It looks like we do [currently include QUnit test files in the JSHint scans](https://github.com/WordPress/wordpress-develop/blob/trunk/Gruntfile.js#L672-L679).
```suggestion
      - 'src/**.php'
      - 'tests/**.php'
      - 'tests/qunit/**/*.js
```

It looks like `Gruntfile.js` specifies the exclusion of `tests/qunit/(vendor|editor)/**` as well. But unless I'm missing something, those directories no longer exist, so there's no need to include these exclusions here.

It also looks like `grunt jshint:tests` uses the `tests/qunit/.jshintrc` file for configuration. I haven't looked into why there are different configurations, but for the time being, below should also be changed to include that file in case it's updated.

TIL that you can't [add comments or suggested changes to unmodified lines in a PR](https://github.com/orgs/community/discussions/4452). So instead here's what I suggest for below.

```
      # This file configures JSHint. Changes could affect the outcome.
      - '**.jshintrc'

---

I also wish that there were an easy way to prevent a specific job from running within a workflow when certain path filters do not match (don't run PHPCS when only `.js` files are touched). However, it seems that can only be accomplished by having a job that detects this (without relying on a third-party action). ChatGPT came up with the following rough example.

```
  check-files:
    runs-on: ubuntu-latest
    outputs:
      changed: ${{ steps.filter.outputs.changed }}
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0
      - id: filter
        run: |
          if git diff --name-only ${{ github.base_ref || github.event.before }} ${{ github.sha }} | grep -qE '\.php$'; then
            echo "changed=true" >> "$GITHUB_OUTPUT"
          else
            echo "changed=false" >> "$GITHUB_OUTPUT"
          fi
```

I don't think there's much benefit in this specific workflow because these jobs complete in under 30 seconds. But if there are any longer-running ones, it may be worthwhile to try.

---

> Misses files in the root directory, currently the PHP sample configs, webpack.

Good catch. PHP files in the root are indeed covered by phpcs so these need re-adding to the path filtering.

The webpack files aren't covered by eslint so they can remain excluded: https://github.com/WordPress/wordpress-develop/blob/cfcb4e9e168b557ea8867ac2b6729c25906a7f54/.eslintignore#L6